hazelcast/src/main/java/com/hazelcast/collection/impl/queue/operations/OfferOperation.java

```diff
@@ -53,23 +53,20 @@ public OfferOperation(final String name, final long timeout, final Data data) {
     @Override
     public void run() {
         QueueContainer queueContainer = getContainer();
+        LocalQueueStatsImpl stats = getQueueService().getLocalQueueStatsImpl(name);
         if (queueContainer.hasEnoughCapacity()) {
             itemId = queueContainer.offer(data);
+            stats.incrementOffers();
             response = true;
         } else {
+            stats.incrementRejectedOffers();
             response = false;
         }
     }
 
     @Override
     public void afterRun() throws Exception {
-        LocalQueueStatsImpl stats = getQueueService().getLocalQueueStatsImpl(name);
-        if (Boolean.TRUE.equals(response)) {
-            stats.incrementOffers();
-            publishEvent(ItemEventType.ADDED, data);
-        } else {
-            stats.incrementRejectedOffers();
-        }
+        publishEvent(ItemEventType.ADDED, data);
```

Comment:
this statement should still be executed on condition `if (Boolean.TRUE.equals(response)) {`

Reply:
Ups, sorry for that. Fixed.